src/main/java/com/klarna/hiverunner/HiveServerContainer.java

```diff
@@ -86,6 +86,7 @@ public void init(Map<String, String> testConfig, Map<String, String> hiveVars) {
         try {
             hiveServer2 = new HiveServer2();
             hiveServer2.init(hiveConf);
+            hiveServer2.start();
```

Comment:
Just curious if this is the cause of those INFO logs that you mentioned over e-mail? I assume this call to `start` is required? Why wasn't it needed in the past?

Reply:
This exception is being thrown without it:

```
java.lang.IllegalStateException: Failed to create HiveServer :Cannot open sessions on an inactive HS2 instance, or the HS2 server leader is not ready; please use service discovery to connect the server leader again

	at com.klarna.hiverunner.HiveServerContainer.init(HiveServerContainer.java:106)
	at com.klarna.hiverunner.builder.HiveShellBase.start(HiveShellBase.java:160)
	at com.klarna.hiverunner.HiveRunnerCore.buildShell(HiveRunnerCore.java:87)
	at com.klarna.hiverunner.HiveRunnerCore.createHiveServerContainer(HiveRunnerCore.java:57)
	at com.klarna.hiverunner.HiveRunnerExtension.createHiveServerContainer(HiveRunnerExtension.java:107)
	at com.klarna.hiverunner.HiveRunnerExtension.postProcessTestInstance(HiveRunnerExtension.java:69)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: org.apache.hive.service.cli.HiveSQLException: Cannot open sessions on an inactive HS2 instance, or the HS2 server leader is not ready; please use service discovery to connect the server leader again
	at org.apache.hive.service.cli.session.SessionManager.createSession(SessionManager.java:436)
	at org.apache.hive.service.cli.session.SessionManager.openSession(SessionManager.java:425)
	at org.apache.hive.service.cli.CLIService.openSession(CLIService.java:211)
	at com.klarna.hiverunner.HiveServerContainer.init(HiveServerContainer.java:100)
	... 19 more
```

I have found this after a very long debugging of a new Hive code-base. I did not check the old one so can not compare.

Reply:
This exception is being thrown without it:

```
java.lang.IllegalStateException: Failed to create HiveServer :Cannot open sessions on an inactive HS2 instance, or the HS2 server leader is not ready; please use service discovery to connect the server leader again

	at com.klarna.hiverunner.HiveServerContainer.init(HiveServerContainer.java:106)
	at com.klarna.hiverunner.builder.HiveShellBase.start(HiveShellBase.java:160)
	at com.klarna.hiverunner.HiveRunnerCore.buildShell(HiveRunnerCore.java:87)
	at com.klarna.hiverunner.HiveRunnerCore.createHiveServerContainer(HiveRunnerCore.java:57)
	at com.klarna.hiverunner.HiveRunnerExtension.createHiveServerContainer(HiveRunnerExtension.java:107)
	at com.klarna.hiverunner.HiveRunnerExtension.postProcessTestInstance(HiveRunnerExtension.java:69)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: org.apache.hive.service.cli.HiveSQLException: Cannot open sessions on an inactive HS2 instance, or the HS2 server leader is not ready; please use service discovery to connect the server leader again
	at org.apache.hive.service.cli.session.SessionManager.createSession(SessionManager.java:436)
	at org.apache.hive.service.cli.session.SessionManager.openSession(SessionManager.java:425)
	at org.apache.hive.service.cli.CLIService.openSession(CLIService.java:211)
	at com.klarna.hiverunner.HiveServerContainer.init(HiveServerContainer.java:100)
	... 19 more
```
I have found this after a very long debugging of a new Hive code-base. I did not check the old one so can not compare.

Reply:
OK, so it sounds like we have to keep the above for now.